rewrite-java/src/main/java/org/openrewrite/java/cleanup/RemoveInstanceOfPatternMatch.java

```diff
@@ -180,7 +169,7 @@ public J.If visitIf(J.If iff, ExecutionContext executionContext) {
         private J.Block addVariableDeclaration(J.Block block, J.InstanceOf instanceOf,
                                                ExecutionContext executionContext) {
             JavaTemplate template = JavaTemplate
-                    .builder(this::getCursor, "#{} #{} = (#{}) #{any()}")
+                    .builder(() -> new Cursor(getCursor(), block), "#{} #{} = (#{}) #{any()};")
```

Comment:
This was the problem here: The cursor didn't match the object being modified with the template. Possibly `JavaTemplate` could issue some kind of warning in these cases. But because of this `JavaTemplate` can then not properly derive the surrounding code required to do proper type attribution of the templated code.

Reply:
I tried to find such an example in the repository. I didn't know that it can be just created. Thanks for the fix! I've disabled the last test. It seems that everything works now!

Reply:
Having to change code in other subtrees than the one being visited is probably also not that common. But there could indeed be more idiomatic ways to solve this problem.